manifests/init.pp

```diff
@@ -36,47 +38,51 @@
   $data_connection_timeout = '120',
   $nopriv_user             = undef,
   $async_abor_enable       = 'NO',
+  $accept_timeout          = '60',
+  $one_process_model       = 'NO',
   $ascii_upload_enable     = 'NO',
   $ascii_download_enable   = 'NO',
   $ftpd_banner             = undef,
   $chroot_local_user       = 'NO',
   $chroot_list_enable      = 'NO',
   $chroot_list_file        = '/etc/vsftpd/chroot_list',
-  $ls_recurse_enable       = 'NO',
+  $secure_chroot_dir       = undef,
+  $ls_recurse_enable       = 'YES',
   $listen                  = 'YES',
   $listen_port             = undef,
+  $log_ftp_protocol        = 'NO',
   $pam_service_name        = 'vsftpd',
   $userlist_enable         = 'YES',
   $userlist_deny           = undef,
   $tcp_wrappers            = 'YES',
   $hide_file               = undef,
   $hide_ids                = 'NO',
-  $setproctitle_enable     = 'NO',
-  $text_userdb_names       = 'NO',
+  $setproctitle_enable     = undef,
```

Comment:
same here with regards to defaults and using hiera. You can specify nil which should equate to undef like this

```yaml
vsftpd::setproctitle_enable: ~
```